services/notification_queue/notification_queue.py

```diff
@@ -666,25 +667,44 @@ async def post(self):
                                     Classification.id == target_id
                                 )
                             ).first()
-                            target_data = target.to_dict()
-                            target_content = source_notification_content(target)
+                            target_data = {
+                                **target.to_dict(),
+                                "group_ids": [group.id for group in target.groups],
+                            }
+                            target_content = source_notification_content(
+                                target, target_type="classification"
+                            )
                         elif is_spectra:
                             users = session.scalars(
                                 sa.select(User).where(
-                                    User.preferences["notifications"][
-                                        "favorite_sources"
-                                    ]["active"]
-                                    .astext.cast(sa.Boolean)
-                                    .is_(True)
+                                    sa.or_(
```

Comment:
Is this better as a get with a default?

Reply:
What do you mean by that?

Reply:
This assumes all those keys exist.

Reply:
I don't think sqlalchemy cares about defaults, I'm not sure it makes a difference. Would be worth looking at the query it generates.
But in this case, where we need the key to be set to a certain value, we do want the user to have all of them anyway.

Maybe your question was just whether or not it yields an error, but we know it doesn't. By default user profiles don't have any key for the notifications altogether, and that hasn't been a problem so far.

Reply:
Here we're looking into the user class, not an actual user returned by a query, so preferences isn't a real dictionary where looking at a missing key creates an exception.